src/Type.h

```diff
@@ -361,7 +363,11 @@ struct Type {
 
     /** Is this type a floating point type (float or double). */
     HALIDE_ALWAYS_INLINE
-    bool is_float() const {return code() == Float;}
+    bool is_float() const {return code() == Float || code() == BFloat;}
```

Comment:
...are we *certain* that this is the right thing to do? Without having examined downstream uses, I'd bet that for some cases we need this semantics but for others we don't. (Alternative is presumably to leave as-is and add `is_float_or_bfloat()`)

Reply:
Yes, I think so. bfloats are still floats, it's just that there are two different ways to make a 16-bit float, so setting bits = 16 is ambiguous. I'm treating them as a subtype.